test/partial-placeholder.civet

```diff
@@ -0,0 +1,142 @@
+{ testCase, throws } from ./helper.civet
+
+describe "partial placeholder", ->
+  testCase """
+    first arg
+    ---
+    f ., a
+    ---
+    $ => f($, a)
+  """
+
+  testCase """
+    second arg
+    ---
+    f a, .
+    ---
+    $ => f(a, $)
+  """
+
+  testCase """
+    both args
+    ---
+    f ., .
+    ---
+    $ => f($, $)
+  """
```

Comment:
As I mentioned above, this is the only example I question. Though I believe I originally suggested this behavior, I wonder whether `($0, $1) => f($0, $1)` would be more useful. Granted, it's no longer an ampersand function this way, so we'd need to use different code when there's more than one argument.

Hmm, thinking more, maybe there are applications for repeated first argument. We're always wondering how to re-use an `&`. You could imagine something like:
```js
f
  key: .
  value: .
```

But I'm not totally sure which is more useful. The behavior you've implemented is more consistent with the future plans for `&`, which is both good (consistency) and bad (different behaviors might collectively cover more cases).

Reply:
I'm leaning towards every `.` reference being the single argument because it combines better with pipelines. Also once we get multiple arguments it is more complex and the benefit is reduced compared to an explicit arrow function (ex. how would one reverse the first and second arguments?). There's definitely room to explore further in the future though.

Reply:
Sounds good. I guess we'll have `...[3]` if we really want. And perhaps more practically, we'll hopefully have `&3`.